src/vm.rs

```diff
@@ -550,6 +532,35 @@ impl<'a, S: Storage> Vm<'a, S> {
             Err(err) => VmExecutionResult::ExecutionError(err),
         }
     }
+
+    // Apply rewards (balance increments) based on self.reward_policy
+    fn apply_rewards(&self, write_set: &mut WriteSet, tx: &TxEnv, gas_used: U256) {
+        let rewards: Vec<(MemoryLocationHash, U256)> = match self.reward_policy {
+            RewardPolicy::Ethereum => {
+                let mut gas_price = if let Some(priority_fee) = tx.gas_priority_fee {
+                    std::cmp::min(tx.gas_price, priority_fee + self.block_env.basefee)
+                } else {
+                    tx.gas_price
+                };
+                if self.spec_id.is_enabled_in(SpecId::LONDON) {
+                    gas_price = gas_price.saturating_sub(self.block_env.basefee);
+                }
+                vec![(self.beneficiary_location_hash, gas_price * gas_used)]
+            }
+        };
+
+        for (recipient, amount) in rewards {
+            if let Some((_, value)) = write_set.iter_mut().find(|entry| entry.0 == recipient) {
+                match value {
+                    MemoryValue::Basic(account_info) => account_info.balance += amount,
```

Comment:
For more consistency with the rest of the code base:
- Use `info` where there is a conversion with `AccountBasic` nearby.
- Use `account` otherwise as the default name.